lib/internal/process/esm_loader.js

```diff
@@ -34,38 +34,52 @@ exports.importModuleDynamicallyCallback = async function(wrap, specifier) {
   throw new ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING();
 };
 
-let ESMLoader = new Loader();
-exports.ESMLoader = ESMLoader;
+const esmLoader = new ESMLoader();
 
+exports.esmLoader = esmLoader;
+
+/**
+ * Causes side-effects: user-defined loader hooks are added to esmLoader.
+ * @returns {void}
+ */
 async function initializeLoader() {
   const { getOptionValue } = require('internal/options');
-  const userLoader = getOptionValue('--experimental-loader');
-  if (!userLoader)
-    return;
+  // customLoaders CURRENTLY can be only 1 (a string)
+  // in the near future, it will be string[]
+  const customLoaders = getOptionValue('--experimental-loader');
+
+  if (!customLoaders.length) return;
+
+  const { emitExperimentalWarning } = require('internal/util');
+  emitExperimentalWarning('--experimental-loader');
+
   let cwd;
   try {
     cwd = process.cwd() + '/';
   } catch {
     cwd = 'file:///';
   }
-  // If --experimental-loader is specified, create a loader with user hooks.
-  // Otherwise create the default loader.
-  const { emitExperimentalWarning } = require('internal/util');
-  emitExperimentalWarning('--experimental-loader');
-  return (async () => {
-    const hooks =
-        await ESMLoader.import(userLoader, pathToFileURL(cwd).href);
-    ESMLoader = new Loader();
-    ESMLoader.hook(hooks);
-    ESMLoader.runGlobalPreloadCode();
-    return exports.ESMLoader = ESMLoader;
-  })();
+
+  // A separate loader instance is necessary to avoid cross-contamination
+  // between internal Node.js and userland
```

Comment:
In docs pages I've seen only `user-land` 🤔
```suggestion
  // between internal Node.js and user-land
```